projectorganizer/src/prjorg-sidebar.c

```diff
@@ -300,6 +300,199 @@ static void on_follow_active(GtkToggleToolButton *button, G_GNUC_UNUSED gpointer
+		locale_path = NULL;
+	}
+
+	/* try to get path from document */
+	if (!locale_path)
+	{
+		GeanyDocument *doc = NULL;
+		doc = document_get_current();
+		if (DOC_VALID(doc))
+			locale_path = g_strdup(doc->real_path);
+	}
+
+	/* Make sure path exists */
+	if(locale_path && !g_file_test(locale_path, G_FILE_TEST_EXISTS))
+	{
+		g_free(locale_path);
+		locale_path = NULL;
+	}
+
+	/* try to get path from project */
+	if (!locale_path && geany_data->app->project)
+	{
+		gchar *utf8_path = NULL;
+		utf8_path = get_project_base_path();
+		locale_path = utils_get_locale_from_utf8(utf8_path);
+		g_free(utf8_path);
+	}
+
+	/* Make sure path exists */
+	if(locale_path && !g_file_test(locale_path, G_FILE_TEST_EXISTS))
+	{
+		g_free(locale_path);
+		locale_path = NULL;
+	}
+
+	return locale_path;
+}
+
+
+/* returns parent folder of path from get_full_path_for_selection()
+ *   if unable to find parent folder, returns home folder
+ *   otherwise, returns NULL
+ * return path is in locale encoding */
+static gchar *get_folder_for_selection(void)
+{
+	gchar *locale_path;
+	locale_path = get_fullpath_for_selection();
+
+	/* Get parent folder if regular file */
+	if (locale_path && g_file_test(locale_path, G_FILE_TEST_IS_REGULAR))
+	{
+		gchar *folder;
+		folder = g_path_get_dirname(locale_path);
+		if (folder && folder == '.')
+		{
+			g_free(folder);
+			folder = NULL;
+		}
+		if (folder && g_file_test(folder, G_FILE_TEST_IS_DIR))
+		{
+			g_free(locale_path);
+			locale_path = folder;
+		}
+		else
+		{
+			g_free(folder);
+			g_free(locale_path);
+			locale_path = NULL;
+		}
+	}
+
+	/* Try home folder */
+	if (!locale_path)
+		locale_path = g_strdup(g_get_home_dir());
+
+	/* Make sure path exists */
+	if(locale_path && !g_file_test(locale_path, G_FILE_TEST_IS_DIR))
+	{
+		g_free(locale_path);
+		locale_path = NULL;
+	}
+
+	return locale_path;
+}
+
+
+void on_open_file_manager(G_GNUC_UNUSED GtkMenuItem * menuitem, G_GNUC_UNUSED gpointer user_data)
+{
+	gchar *locale_path = NULL;
+	static gchar *open_command = NULL;
+
+	if (!open_command)
+		open_command = g_strdup(PRJORG_COMMAND_OPEN);
+
+	if (prj_org)
+	{
+		g_free(open_command);
+		open_command = g_strdup(prj_org->command_file_manager);
+	}
+
+	locale_path = get_folder_for_selection();
+	if (locale_path)
+	{
+		gchar *command;
+		command = g_strconcat (open_command, " \"", locale_path, "\"", NULL);
+		if (!spawn_async(locale_path, command, NULL, NULL, NULL, NULL))
+			msgwin_status_add("Unable to open folder.  Command unavailable: %s", open_command);
+
+		g_free(command);
+		g_free(locale_path);
+	}
+	else
+	{
+		msgwin_status_add("Unable to find folder");
+	}
+}
+
+
+void on_open_terminal(G_GNUC_UNUSED GtkMenuItem * menuitem, G_GNUC_UNUSED gpointer user_data)
+{
+	gchar *locale_path = NULL;
+	static gchar *open_command = NULL;
+
+	if (!open_command)
+		open_command = g_strdup(PRJORG_COMMAND_TERMINAL);
+
+	if (prj_org)
+	{
+		g_free(open_command);
+		open_command = g_strdup(prj_org->command_terminal);
+	}
```

Comment:
Personally I'd just avoid the static variable and would do
```
open_command = prj_org ? g_strdup(prj_org->command_terminal) : g_strdup(PRJORG_COMMAND_TERMINAL);
```
and would free this at the end of the function.

Reply:
In fact, the g_strdup() seems to be unnecessary so it could be just
```
open_command = prj_org ?prj_org->command_terminal : PRJORG_COMMAND_TERMINAL;
```